framework/src/main/java/org/checkerframework/framework/type/AbstractViewpointAdapter.java

```diff
@@ -408,6 +414,23 @@ private AnnotatedTypeMirror substituteTVars(AnnotatedTypeMirror lhs, AnnotatedTy
             // Base case where actual type argument is extracted
             if (lhs.getKind() == TypeKind.DECLARED) {
                 rhs = getTypeVariableSubstitution((AnnotatedDeclaredType) lhs, atv);
```

Comment:
Looking into whether we can add this change to the method.

Reply:
@Ao-senXiong This is still open.

Reply:
I have added clearer comment and make the test case nicer.